spec/unit/request_spec.rb

```diff
@@ -128,52 +128,96 @@
 
     it "doesn't overwrite user and password (which may have already been set by the Resource constructor) if there is no user/password in the url" do
       URI.stub(:parse).and_return(double('uri', user: nil, password: nil, hostname: 'example.com'))
-      @request = RestClient::Request.new(:method => 'get', :url => 'example.com', :user => 'beth', :password => 'pass2')
-      @request.send(:parse_url_with_auth!, 'http://example.com/resource')
-      @request.user.should eq 'beth'
-      @request.password.should eq 'pass2'
```

Comment:
Unrelated, but I prefer to scope variables as tightly as possible.  Local variable `request` is more restrictive than instance variable `@request`.